### PR TITLE
MULE-10580: Operation reconnection should be decoupled from config re…

### PR DESCRIPTION
**Pull request by marianogonzalez**
…connection

### CODE REVIEW

File: modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/resources/ExtensionModelJsonGeneratorTestCase.java [hunk 1]
```diff
@@ -152,8 +155,24 @@ public void generate() throws Exception {
...
+      if (fixInputFiles) {
+        File root = new File(getResourceAsUrl("models/" + expectedSource, getClass()).toURI()).getParentFile()
+            .getParentFile().getParentFile().getParentFile();
+        File testDir = new File(root, "src/test/resources/models");
+        File target = new File(testDir, expectedSource);
+        stringToFile(target.getAbsolutePath(), json);
+
+        System.out.println(expectedSource + " fixed");
+      } else {
+        System.out.println("Expected: \n " + expectedJson);
```
Comment (pablolagreca) [inline modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/resources/ExtensionModelJsonGeneratorTestCase.java]: Shouldn't this be done with a logger?

File: modules/extensions-spring-support/src/test/resources/META-INF/documentation-extension-descriptions.xml [hunk 2]
```diff
@@ -31,24 +31,24 @@
         <connection name="another">
             <description><![CDATA[Another Provider Documentation]]></description>
             <parameters>
-                <parameter name="reconnectionStrategy">
-                    <description><![CDATA[A retry strategy in case of connectivity errors]]></description>
+                <parameter name="reconnection">
+                    <description><![CDATA[TODO]]></description>
```
Comment (pablolagreca) [inline modules/extensions-spring-support/src/test/resources/META-INF/documentation-extension-descriptions.xml]: fill the TODO (more in the same file below)

File: modules/extensions-spring-support/src/test/resources/models/heisenberg.json [hunk 3]
```diff
@@ -4083,7 +4370,7 @@
                         "annotations": {
                           "typeId": "java.lang.Object",
                           "typeAlias": {
-                            "value": "repeatable-file-store-stream"
+                            "value": "repeatable-file-store-iterable"
```
Comment (pablolagreca) [inline modules/extensions-spring-support/src/test/resources/models/heisenberg.json]: why is this change here?

File: modules/spring-config/src/main/resources/META-INF/mule.xsd [hunk 4]
```diff
@@ -1384,6 +1384,28 @@
     </xsd:element>
     <xsd:complexType name="abstractReconnectNotifierType"/>
 
+    <xsd:element name="reconnection" type="reconnectionConfigType">
+        <xsd:annotation>
+            <xsd:documentation>
+                TODO
```
Comment (pablolagreca) [inline modules/spring-config/src/main/resources/META-INF/mule.xsd]: Something is missing here

File: modules/spring-config/src/main/java/org/mule/runtime/config/spring/internal/dsl/model/InfrastructureElementModelDelegate.java [hunk 5]
```diff
@@ -155,8 +155,50 @@ public void visitObjectValue(ParameterObjectValue objectValue) {
...
+
+    final ComponentConfiguration result = config.build();
+    parentConfig.withNestedComponent(result);
+    parentElement.containing(elementBuilder.withConfig(result).build());
+
+  }
+
   private void createReconnectionStrategy(ParameterValue value,
-                                          ParameterModel parameterModel,
+                                          Object parameterModel,
```
Comment (pablolagreca) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/internal/dsl/model/InfrastructureElementModelDelegate.java]: Why Object?

File: core/src/main/java/org/mule/runtime/core/api/extension/MuleExtensionModelDeclarer.java [hunk 6]
```diff
@@ -360,21 +351,6 @@ private void declareScatterGather(ExtensionDeclarer extensionDeclarer, ClassType
         .defaultingTo(Integer.MAX_VALUE)
         .withExpressionSupport(NOT_SUPPORTED)
         .describedAs("This value determines the maximum level of parallelism that will be used by this router. .");
-    scatterGather.onDefaultParameterGroup()
```
Comment (marianogonzalez) [inline core/src/main/java/org/mule/runtime/core/api/extension/MuleExtensionModelDeclarer.java]: yes. piggy backing. ExtensionModelFactory will add this automatically

File: modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/resources/ExtensionModelJsonGeneratorTestCase.java (hunk 1 above)
Comment (marianogonzalez) [inline modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/resources/ExtensionModelJsonGeneratorTestCase.java]: we do it like this so that we can easily copy paste for comparing

File: modules/extensions-spring-support/src/test/resources/models/heisenberg.json (hunk 3 above)
Comment (marianogonzalez) [inline modules/extensions-spring-support/src/test/resources/models/heisenberg.json]: the alias changed.. piggy backing. I will work more on this on MULE-13247

File: modules/spring-config/src/main/java/org/mule/runtime/config/spring/internal/dsl/model/InfrastructureElementModelDelegate.java (hunk 5 above)
Comment (marianogonzalez) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/internal/dsl/model/InfrastructureElementModelDelegate.java]: technical limitations I already discussed with Marra and will have to tackle separately. Basically this can either be a parameterModel or a declaration

Comment (pablolagreca) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/internal/dsl/model/InfrastructureElementModelDelegate.java]: add TODO
